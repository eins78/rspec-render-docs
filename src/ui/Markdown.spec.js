/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'

import { Md } from './Markdown'

const inlineMarkdownExamples = [
  'plain text',
  '*expected* **inline** `markdown` ***content*** *`yay`*',
  `
content with blocks like

### Block-Level Heading

> and other ***non-inline*** markdown

is not allowed
==============

but will be output as plain text
  `,
  `any <em>html tags</em> are stripped as well`
]

inlineMarkdownExamples.forEach((str, i) => {
  it(`renders example ${i + 1}`, () => {
    const component = renderer.create(<Md>{str}</Md>)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
