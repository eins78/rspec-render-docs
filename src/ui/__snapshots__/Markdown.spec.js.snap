exports[`test renders example 1 1`] = `
<div>
  plain text
</div>
`;

exports[`test renders example 2 1`] = `
<div>
  <em>
    expected
  </em>
   
  <strong>
    inline
  </strong>
   
  <code
    data-sourcepos={undefined}>
    markdown
  </code>
   
  <strong>
    <em>
      content
    </em>
  </strong>
   
  <em>
    <code
      data-sourcepos={undefined}>
      yay
    </code>
  </em>
</div>
`;

exports[`test renders example 3 1`] = `
<div>
  content with blocks like
  Block-Level Heading
  and other 
  <strong>
    <em>
      non-inline
    </em>
  </strong>
   markdown
  is not allowed
  but will be output as plain text
</div>
`;

exports[`test renders example 4 1`] = `
<div>
  any 
  html tags
   are stripped as well
</div>
`;
